### PR TITLE
update rasa-sdk to support python 3.11, 3.12

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Add Python 3.11 and 3.12 support, upgrade dependencies

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation in the [rasaHQ/rasa](https://github.com/rasaHQ/rasa)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [ ] reformat files using `ruff` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -212,8 +212,8 @@ def test_cli_invalid_format_value_in_config(caplog: LogCaptureFixture) -> None:
 
 
 @pytest.mark.skipif(
-    sys.version_info.minor in [9, 10],
-    reason="no error is raised with python 3.9 or 3.10",
+    sys.version_info.minor != 8,
```
[the remark this answers is not in the document]
Comment (ancalita) [reply tests/test_utils.py:215]: Shouldn't we remove support for 3.8 actually if that's been dropped on rasa pro side?

[the remark this answers is not in the document]
Comment (vcidst) [reply tests/test_utils.py:215]: I think yeah we should, not sure why it wasn't done. Let me do that too

File: rasa_sdk/grpc_py/action_webhook_pb2.py
[the remark this answers is not in the document]
Comment (vcidst) [reply rasa_sdk/grpc_py/action_webhook_pb2.py:1]: `make check-generate-grpc-code-in-sync` is part of `make lint` which failed the CI until I generated the gRPC code and committed it

[the remark this answers is not in the document]
Comment (ancalita) [reply rasa_sdk/grpc_py/action_webhook_pb2.py:1]: Got it, thanks!

File: tests/test_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vcidst) [reply tests/test_utils.py]: Done, I've removed Python 3.8 support

**Pull request merged**